screens/Schedule.js

```diff
@@ -107,12 +193,40 @@ export default class Schedule extends Component<Props> {
       <View key={index}>
         <ScheduleCard
           item={item}
-          onClick={this.onSelectEvent}
+          onClick={() =>
+            this.toggleModal(
+              item.title,
+              item.eventType,
+              item.desc,
+              item.tags,
+              item.start,
+              item.end,
+              item.restaurant_name,
+              item.restaurant_link,
+              item.menu
+            )
+          }
           title={item.title}
           tags={item.tags}
         >
           {item.desc}
         </ScheduleCard>
+        <Modal isVisible={this.state.isModalVisible}>
+          <Event
+            isModalVisible={() =>
+              this.toggleModal(null, null, null, null, null, null, null, null)
+            }
+            title={this.state.modalTitle}
```

Comment:
I feel like you can set the component you're rendering itself in `setState`, which may be cleaner here. Interesting pattern choice here